### DISCUSSION TITLE
Mealie v1 Beta Release

### DISCUSSION THREAD

**Discussion by hay-kot** [mealie-recipes/mealie#1073] [General]
# Mealie v1 Beta Release

Heyo Folk! I'm super excited to finally be able to say that the Mealie v1 Beta is out and ready for testing! As a reminder, this beta includes HUGE improvements and nearly an entire rewrite of the Frontend and Backend Code. It's been nearly 6 months since I started down the path of rewriting the Frontend that's spun into nearly a full rewrite with a whole host of new features. For a full set of features checkout the release notes.

- [Mealie v1 Beta Release Notes](https://nightly.mealie.io/changelog/v1.0.0/)

v1 Is still an on going project, if you're interested in tracking progress you can use our [project board](https://github.com/hay-kot/mealie/projects/7).

## What I Need From You!

As apart of the Beta release process, I need you to help me test and validate the new features and code. To deploy Mealie v1 Beta you'll need to follow the instructions on the [nightly documentation](https://nightly.mealie.io) and the [migration guide](https://nightly.mealie.io/documentation/getting-started/migrating-to-mealie-v1/). At this time it is recommended that you run the beta along side the latest version of Mealie, however many users (including myself) have completely switched over to beta. 

### Before you Deploy

Before you deploy there are a few things you should be aware of

- Multilingual support is limited. We're still in the process of migration our translation efforts to the new version, but we're hoping to wrap this up soon.
- Mealie now requires 2 docker containers to be running. One for the frontend and one for the backend. The documentation does a good job of outlining the requirements, but using docker-compose you should be able to easily deploy the containers. 
- Beta versions will be pushed to the `mealie:frontend-nightly` and `mealie:api-nightly` tags. You should disable automated updates and be thoughtful about when you pull new images.

### Reporting Issues

If you've found an issue in the beta you can file a bug report on GitHub, just be sure to use the v1 template so we know what version it's for. Even better, if you're a developer you can open a pull requests that fixes the issue.

### User Experience

Have strong feelings about one of the new UI flows? Let me know! If it's a minor gripe, or a celebration feel free to comment in the section below and provide any details on your experience. If it's a larger issue that requires mediation or a larger discussion feel free to open a new issue on GitHub.

## What's Next?

Well, first things first we need to get v1 out the door. I'm excited to see what you guys think of the new features and code. Once we can get to a stable release my plan is to start working on a Native UI application with a minimal set of features. Mainly I want to get the shopping lists and a recipe viewer and continue on from there. 

It's a lofty goal but I'm excited to work on it! One of the major barriers to this is finding additional contributors to help me maintain the Mealie server and frontend. If you're interested in becoming a contributor and committing to taking on a role to resolve issues and track down bugs, please reach out on Discord or GitHub and we can work get that process started!

**Comment by FonduemangVI** [comment]
Okay so I might be doing it wrong but Mealie v1 is reporting that email is not ready yet even though I have set all the enviro variables.

**Reply by hay-kot** [reply]
Hey, all that does is check that the following ENV variables aren't an empty string or None in the backend container - so you'll want to make sure all of these are defined.

```
SMTP_HOST
SMTP_PORT
SMTP_FROM_NAME
SMTP_TLS
SMTP_FROM_EMAIL
SMTP_USER
SMTP_PASSWORD
```

**Reply by FonduemangVI** [reply]
Right okay. So I am using an SMTP Relay and don't need a username or password. What do I put into the SMTP_USER and
SMTP_PASSWORD variables?

Apologies if I am misunderstanding

**Reply by hay-kot** [reply]
That isn't something I've tested so I'm not sure. You could try just putting in a "-"  as a place holder and see if it works?

If you can get your relay working submit an issue and I'll look into it.

**Reply by FonduemangVI** [reply]
Sorry for the delayed reply. the "-" works and Mealie will let me test sending an email. I get an error though "send email error: [Errno None] SMTP AUTH extension not supported by server.: None"
I am trying to use office 365

**Reply by FonduemangVI** [reply]
Any thoughts on why I am getting the error?

**Comment by brettinternet** [comment]
Hi @hay-kot, thank you for your wonderful work with this release.

Is there a reason you chose to serve the client with node rather than just serving the Vue app directly from the python server? I wonder if serving the app from a single server would help simplify the architecture - for both development and production.

**Reply by hay-kot** [reply]
Server Side Render and Generated Social cards are impossible without the Node application. 

I also don't think that having two containers is really that complicated. The documentation provides a docker-compose example, and a explanation of the required configuration if someone desires to make changes to the deployment. 

The project also provides a dev container  that bundles all the setup for you so you just have to start the servers after you open VS Code.

**Reply by brettinternet** [reply]
Oh your Vue app is SSR for the meta tags. I see.
